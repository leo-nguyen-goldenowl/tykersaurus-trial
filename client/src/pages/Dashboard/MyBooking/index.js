import React from 'react'

const MyBooking = () => {
  return (
    <div className='my-booking'>
      <table>
        <thead>
          <tr>
            <th>Date</th>
            <th>Tee time</th>
            <th>Quantity of Players</th>
            <th>Start Tee</th>
            <th>Golf Course</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>29 December 2020</td>
            <td>17:43</td>
            <td>4</td>
            <td>
              1<sup>st</sup>
            </td>
            <td>Classic course</td>
          </tr>
        </tbody>
      </table>
    </div>
  )
}

export default MyBooking
