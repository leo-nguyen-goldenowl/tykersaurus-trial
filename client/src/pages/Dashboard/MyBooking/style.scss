@import "src/assets/scss/_variables.scss";
@import "src/assets/scss/_mixins.scss";
@import "src/assets/scss/_extends.scss";

.my-booking {
  width: 100%;
  &__container {
    width: 100%;
    max-height: 505px;
    overflow-y: auto;
    padding-top: 12px;
  }
  &__title {
    @include text(SFProDisplay-Bold, 16px, $midnight-moss, null);
  }
  table {
    border-collapse: separate;
    border-spacing: 0 6px;
    width: calc(100% - 1px);
    .hide {
      opacity: 0;
      // display: none;
    }
    th,
    td {
      text-align: center;
      &:first-child {
        padding: 20px 0 20px 30px;
        text-align: left;
      }
      &:last-child {
        padding: 20px 20px 20px 0;
        text-align: right;
      }
    }
    thead {
      th {
        font-family: SFProDisplay-Regular;
        font-size: 16px;
        color: $dim-gray;
        font-weight: 400;
      }
    }
    tbody {
      tr {
        td {
          position: relative;
          font-family: SFProDisplay-Medium;
          font-size: 16px;
          color: $midnight-moss;
          background: $white;
          &::before {
            content: "";
            position: absolute;
            background-color: blue;
            width: 100%;
            height: 100%;
            left: 0;
            top: -1px;
            border-top: 1px solid transparent;
            border-bottom: 1px solid transparent;
            background: transparent;
          }
          &:first-child {
            border-top-left-radius: 8px;
            border-bottom-left-radius: 8px;
            &::before {
              border-left: 1px solid transparent;

              border-top-left-radius: 8px;
              border-bottom-left-radius: 8px;
            }
          }
          &:last-child {
            border-top-right-radius: 8px;
            border-bottom-right-radius: 8px;
            &::before {
              border-top-right-radius: 8px;
              border-right: 1px solid transparent;
              border-bottom-right-radius: 8px;
            }
          }
        }

        &:hover {
          td {
            &::before {
              border-color: $light-grey;
            }
          }
        }
      }
    }
  }
  table.main {
    margin-top: -82px;
    @media only screen and (min-width: 601px) and (max-width: 700px) {
      margin-top: -101px;
    }
  }
  &__tee-time {
    width:20%;
  }
  &__player {
    width: 25%;
  }
  &__start {
    width: 15%;
  }
  &__course {
    width: 20%;
  }
}
