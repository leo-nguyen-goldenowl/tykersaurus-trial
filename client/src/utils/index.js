export const getValueFromStorage = (name) => {
  return localStorage.getItem(name)
}
