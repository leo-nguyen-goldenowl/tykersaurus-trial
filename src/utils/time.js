const moment = require('moment')

const convertTeeTimeToMinute = (teeTime) => {
  const listPartOfTeeTime = teeTime.split(':')
  const hourTeeTime = listPartOfTeeTime[0]
  const minuteTeeTime = listPartOfTeeTime[1]

  return Number(hourTeeTime * 60 + minuteTeeTime)
}

const getMaxDate = () => {
  const todayMoment = moment()
  return moment(todayMoment).add(
    moment(moment()).diff(
      moment(
        moment().add(-1, 'days').format('MM/DD/YYYY 23:00'),
        'MM-DD-YYYY hh:mm'
      ),
      'seconds'
    ) > 0
      ? 7
      : 6,
    'days'
  )
}

const accessBooking = (date) => {
  const maxDate = getMaxDate()

  return (
    moment(moment(maxDate).format('MM/DD/YYYY 00:00')).diff(
      moment(date),
      'seconds'
    ) >= 0
  )
}

module.exports = { convertTeeTimeToMinute, accessBooking, getMaxDate }
