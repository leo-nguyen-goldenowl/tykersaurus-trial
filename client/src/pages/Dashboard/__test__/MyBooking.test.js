import { testSnapshots } from "utils/test/test.utils";
import MyBooking from "../MyBooking";

jest.mock("react-redux", () => ({
  useDispatch: () => (args) => args,
  useSelector: () => (args) => args,
}));
// .mock("redux/services/receipt", () => ({
//   fetchListReceiptIfNeeded: () => (args) => args,
// }))

describe("Test MyBooking component", () => {
  testSnapshots(MyBooking, [
    {
      props: {},
      description: "MyBooking component should match snapshots",
    },
  ]);
});
