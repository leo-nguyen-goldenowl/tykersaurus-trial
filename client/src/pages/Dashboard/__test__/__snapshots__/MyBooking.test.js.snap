// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MyBooking component snapshots MyBooking component should match snapshots 1`] = `
<div
  className="my-booking"
>
  <p
    className="my-booking__title"
  >
    My booking
  </p>
  <table>
    <thead>
      <tr>
        <th
          className="my-booking__date"
        >
          Date
        </th>
        <th
          className="my-booking__tee-time"
        >
          Tee time
        </th>
        <th
          className="my-booking__player"
        >
          Quantity of Players
        </th>
        <th
          className="my-booking__course"
        >
          Golf Course
        </th>
      </tr>
    </thead>
  </table>
  <div
    className="my-booking__container"
  >
    <table
      className="main"
    >
      <thead
        className="hide"
      >
        <tr>
          <th
            className="my-booking__date"
          >
            Date
          </th>
          <th
            className="my-booking__tee-time"
          >
            Tee time
          </th>
          <th
            className="my-booking__player"
          >
            Quantity of Players
          </th>
          <th
            className="my-booking__course"
          >
            Golf Course
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
    <p>
      Do not have any booking course
    </p>
  </div>
</div>
`;
